docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -482,6 +796,86 @@ This section talks about updating a cluster which was created using a `ClusterCl
 
 ![Update cluster with ClusterClass](./images/cluster-class/update.png)
 
+#### Behavior with patches
+
+This section highlights how the above discussed basic behavior changes when patches are used. This is an important use case because without 
+patches apart from a few properties every Cluster of a ClusterClass would be the same. Patches are used to customize individual Clusters, 
+to avoid creating separate ClusterClasses for every small deviation, e.g. a different HTTP proxy configuration.
+
+##### Create a new ClusterClass with patches
+
+1. User creates a ClusterClass object with variables and patches (other fields are omitted for brevity).
+   ```yaml
+   apiVersion: cluster.x-k8s.io/v1alpha4
+   kind: ClusterClass
+   metadata:
+     name: my-cluster-class
+   spec:
+     [...]
+     variables:
+       definitions:
+       - name: region
+         required: true
+         schema:
+           openAPIV3Schema:
+             type: string
+       - name: controlPlaneMachineType
+         schema:
+           openAPIV3Schema:
+             type: string
+             default: t3.large
+     patches:
+     - name: region
+       definitions:
+       - selector:
+           apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+           kind: AWSClusterTemplate
+         jsonPatches:
+         - op: replace
+           path: “/spec/template/spec/region”
+           valueFrom:
+             variable: region
+     - name: controlPlaneMachineType
+       definitions:
+       - selector:
+           apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+           kind: AWSMachineTemplate
+           matchResources:
+             controlPlane: true
+         jsonPatches:
+         - op: replace
+           path: “/spec/template/spec/instanceType”
+           valueFrom:
+             variable: machineType
+   ```
+
+##### Create a new Cluster with patches
+
+1. User creates a Cluster referencing the ClusterClass created above and defining variables (other fields are omitted for brevity).
+   ```yaml
+   apiVersion: cluster.x-k8s.io/v1alpha4
+   kind: Cluster
+   metadata:
+     name: my-cluster
+   spec:
+     topology:
+       class: my-cluster-class
+       [...]
+       variables:
+       - name: region
+         value: us-east-1
+   ```
+   **Note**: `controlPlaneMachineType` will be defaulted to `t3.large` through a mutating webhook.
```

Comment:
can you expand on this defaulting mechanism? In this case it's defaulted to `t3.large` because that's the default value defined in the example ClusterClass, not because that's an arbitrary default value defined as part of the proposal (might not be obvious to all readers)